blaze-core/src/main/scala/org/http4s/blaze/Http1Stage.scala

```diff
@@ -234,22 +238,23 @@ object Http1Stage {
     else CachedEmptyBufferThunk
   }
 
-  /** Encodes the headers into the Writer, except the Transfer-Encoding header which may be returned
+  /** Encodes the headers into the Writer. Does not encode `Transfer-Encoding` or
+    * `Content-Length` headers, which are left for the body encoder. Adds
+    * `Date` header if one is missing and this is a server response.
+    *
     * Note: this method is very niche but useful for both server and client. */
-  def encodeHeaders(headers: Headers, rr: Writer, isServer: Boolean): Option[`Transfer-Encoding`] = {
-    var encoding: Option[`Transfer-Encoding`] = None
+  def encodeHeaders(headers: Iterable[Header], rr: Writer, isServer: Boolean) = {
```

Comment:
I'm doing this in my head instead of in MiMa, but if it continued to return the `Option[Transfer-Encoding]`, I think it would be fully binary compatible for release in 0.12.4.

As for 0.13, I'm speculating it was written this way originally to avoid an immediate second traversal of the headers in `Http1ServerStage`.  Now that microoptimization is uglier, because `Content-Length` also became important.  I think it's clearer how it's written now, but I don't have a sense as to whether this is an important optimization or a premature one.